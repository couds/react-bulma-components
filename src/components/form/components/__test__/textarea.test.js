import React from 'react';
import renderer from 'react-test-renderer';
import Textarea from '../textarea';

describe('Textarea component', () => {
  it('Should exist', () => {
    expect(Textarea).toMatchSnapshot();
  });
  it('Should have textarea classname', () => {
    const component = renderer.create(<Textarea />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should concat classname in props with Bulma classname', () => {
    const component = renderer.create(
      <Textarea className="other-class this-is-a-test" />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should use inline styles', () => {
    const component = renderer.create(<Textarea style={{ height: 250 }} />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be large with 10 rows and readOnly', () => {
    const component = renderer.create(
      <Textarea rows={10} readOnly size="large" />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('Should be disabled with placeholder and value', () => {
    const component = renderer.create(
      <Textarea value="TEST" disabled placeholder="hello tests" />,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
