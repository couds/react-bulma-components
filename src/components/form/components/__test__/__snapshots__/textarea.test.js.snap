// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea component Should be disabled with placeholder and value 1`] = `
<textarea
  className="textarea"
  disabled={true}
  name=""
  placeholder="hello tests"
  readOnly={false}
  rows={4}
  style={Object {}}
  value="TEST"
/>
`;

exports[`Textarea component Should be large with 10 rows and readOnly 1`] = `
<textarea
  className="textarea is-large"
  disabled={false}
  name=""
  placeholder=""
  readOnly={true}
  rows={10}
  style={Object {}}
  value=""
/>
`;

exports[`Textarea component Should concat classname in props with Bulma classname 1`] = `
<textarea
  className="textarea other-class this-is-a-test"
  disabled={false}
  name=""
  placeholder=""
  readOnly={false}
  rows={4}
  style={Object {}}
  value=""
/>
`;

exports[`Textarea component Should exist 1`] = `[Function]`;

exports[`Textarea component Should have textarea classname 1`] = `
<textarea
  className="textarea"
  disabled={false}
  name=""
  placeholder=""
  readOnly={false}
  rows={4}
  style={Object {}}
  value=""
/>
`;

exports[`Textarea component Should use inline styles 1`] = `
<textarea
  className="textarea"
  disabled={false}
  name=""
  placeholder=""
  readOnly={false}
  rows={4}
  style={
    Object {
      "height": 250,
    }
  }
  value=""
/>
`;
